### PR TITLE
fix: prevent git CLI to prompt user/password on CI

### PR DESCRIPTION
**Pull request by pvdlg**
Per https://github.com/semantic-release/semantic-release/issues/699#issuecomment-373445979

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -22,6 +22,10 @@ async function run(options, plugins) {
   if (!isCi && !options.dryRun && !options.noCi) {
     logger.log('This run was not triggered in a known CI environment, running in dry-run mode.');
     options.dryRun = true;
+  } else {
+    // When running on CI, prevent the `git` CLI to prompt for username/password
```
Comment (gr2m) [inline index.js]: I would suggest we add the link to this PR here for easier reference in future: https://github.com/semantic-release/semantic-release/issues/699#issuecomment-373445979

Comment (pvdlg) [reply index.js]: Should the comment and [GitHub git blame](https://help.github.com/articles/tracing-changes-in-a-file/) be enough?
Personally I use Git blame really often, but not sure if it's a well known feature.

I'm not sure if in general it's better to reference PR/issues as comment versus using GitHub/Git feature. Or at least I don't know in which cases it worst it to reference an PR/issue in a comment. Any suggestion?

**Comment by AVVS** [comment]
@pvdlg @gr2m It doesnt seem to solve the issue

I'm not sure if its some sort of weirdness with env not being passed down with execa, but thats what I get when manually testing, notice that when I pass env variables manually, its all good, but process.env thats been modified inside the nodejs process is actually not being treated by execa anyhow. Also the timeout option itself doesnt trigger for some reason

screen shot 2018-03-18 at 9 28 13 pm

**Comment by gr2m** [comment]
Could you please create a follow up issue? It will be easier for us to track

**Comment by AVVS** [comment]
@gr2m done :)

**Pull request merged**